### PR TITLE
Update list of supported JVMs

### LINKED ISSUE #797: Supported JVMs section needs to be clarified
The `Supported JVMs` section in the Reference Manual is a bit confusing for supported OpenJDK.

https://docs.hazelcast.org/docs/latest/manual/html-single/index.html#supported-jvms

I see that in 4.0 BETA-2 Reference Manual has been updated with the `OpenJDK` column but it may also include a piece of information for JDK version `12` and `13` if they are supported or not.

Also, there is a note (like `JDK not available yet`) under the `Oracle JDK` column for version `11`. Same in the new Reference Manual as well: https://docs.hazelcast.org/docs/4.0-BETA-2/manual/html-single/index.html#supported-jvms

### PR DESCRIPTION
**Pull request by JakeSCahill**
Fixes #797 

- Reformatted the table to list only supported JDKs and their versions
- Added a note with a link to previous manuals for supported JDKs of older versions of Hazelcast
- Fixed the broken link to the TLS/SSL section

### CODE REVIEW

**Review by Serdaro — APPROVED** [review]

File: src/docs/asciidoc/installing_upgrading.adoc [hunk 1]
```diff
@@ -44,7 +44,7 @@ of the link:https://github.com/hazelcast/hazelcast-command-line[Command Line Int
 === Maven
 
 NOTE: As a prerequisite, make sure you have Java installed on your system.
-If you're using JDK 9 and newer, see <<running-in-modular-java>>.
+If you're using JDK 11 or later, see <<running-in-modular-java>>.
```
Comment (Holmistr) [inline src/docs/asciidoc/installing_upgrading.adoc:47]: Why did you change the JDK version to 11? It's actually true of the 9.

Comment (Holmistr) [reply src/docs/asciidoc/installing_upgrading.adoc:47]: Ah I see. We now support only the LTS releases. Therefore, the Java 9 version in the "Running in Modular Java" should be changed as well.

File: src/docs/asciidoc/installing_upgrading.adoc [hunk 2]
```diff
@@ -543,7 +543,7 @@ It is worth mentioning that a business app upgrade is orthogonal to a rolling me
 === Running in Modular Java
 
 Java link:http://openjdk.java.net/projects/jigsaw/[project Jigsaw^] brought
-a new Module System into Java 9 and newer. Hazelcast supports running in
+a new Module System into Java 9 and later. Hazelcast supports running in
```
Comment (Holmistr) [inline src/docs/asciidoc/installing_upgrading.adoc:546]: Probably want to update to Java 11 to be consistent.

**Review by Holmistr — COMMENTED** [review]
Minor comments.

File: src/docs/asciidoc/installing_upgrading.adoc (hunk 2 above)
Comment (JakeSCahill) [reply src/docs/asciidoc/installing_upgrading.adoc:546]: Updating this to Java 11 would make it incorrect. Maybe we can say `If you're using JDK 11 or later, you can use Hazelcast as a module in the [Java Platform Module System](link to more info).`

**Comment by Serdaro** [comment]
@Holmistr , if you are also okay with Jake's latest content suggestion above , could you please approve the PR?

**Review by Holmistr — APPROVED** [review]
